src/main/java/com/michelin/ns4kafka/controllers/SchemaController.java

```diff
@@ -100,52 +101,63 @@ public Mono<HttpResponse<Schema>> apply(String namespace, @Valid @Body Schema sc
         }
 
         return schemaService
-            .validateSchemaCompatibility(ns.getMetadata().getCluster(), schema)
-            .flatMap(validationErrors -> {
-                if (!validationErrors.isEmpty()) {
-                    return Mono.error(new ResourceValidationException(validationErrors, schema.getKind(),
-                        schema.getMetadata().getName()));
-                }
+                .getLatestSubject(ns, schema.getMetadata().getName())
+                .map(Optional::of)
+                .defaultIfEmpty(Optional.empty())
+                .flatMap(latestSubjectOptional -> {
+                    if (latestSubjectOptional.isPresent()) {
+                        var newSchema = new AvroSchema(schema.getSpec().getSchema());
```

Comment:
There's a constructor for `AvroSchema` that takes schema references, here you take a risk to compare without references later in `canonicalString()`. 
It was done automatically on the registry side before